app.js

```diff
@@ -19,15 +19,23 @@ var bot = Slackbot(require('@slack/client'), env);
 var tweeter = TwitterPoster(env);
 var log = helpers.logger('App', env.isDev);
 
+function onImageDataDone (err, imageData, trimmedText) {
+  if (err) {
+    throw err;
+  }
+  console.log('callback called!');
+  tweeter.postTweetAndImage(imageData, trimmedText);
+}
+
 function postText (message) {
+  console.log(message);
   var text = message.item.message && message.item.message.text;
   if (text && text.length <= 140) {
     tweeter.postTweet(text);
   } else if (text && text.length > 140) {
     log('generating tweet for text > 140 characters');
     var trimmedText = text.substring(0, 110) + '…';
-    var imageData = createFakeScreenshot(text);
-    tweeter.postTweetAndImage(imageData, trimmedText);
+    createFakeScreenshot(message.item.message.permalink, trimmedText, onImageDataDone);
```

Comment:
first of all, does `createFakeScreenshot` actually need `trimmedText` anymore? all it needs is the permalink and the callback, right? so it should just be `createFakeScreenshot(message.item.message.permalink, onImageDataDone)`.

since you already have the message here as `trimmedText`, you should just declare `onImageDataDone` inside the `postText` function, so it has access to `trimmedText` in the closure. you could even declare it as an anonymous function inline in the `createFakeScreenshot` call, which would be the most "idiomatic" way to do it.
```js
createFakeScreenshot(message.item.message.permalink, helpers.handleError((imageData) => tweeter.postTweetAndImage(imageData, trimmedText))
```